### PR TITLE
fix(searchList): speed up ctrl-k search by using SPFM

### PR DESCRIPTION
**Pull request by jrainville**
### What does the PR do

Speeds up the CTRL+K search popup search by using a SPFM instead of making objects invisible.

### Affected areas

The CTRL+K popup only

### Architecture compliance

- [C] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Architecture guidelines](../architecture-guide.md)

### Screenshot of functionality (including design for comparison)

[speed-ctrl-k.webm](https://github.com/user-attachments/assets/7f8e8931-5b16-4579-b531-3fce88aaee5d)

### Impact on end user

Makes using the ctrl-K menu way more snappy

### How to test

- Open tke ctrl-k popup and search channels

### Risk 

Tick **one**:
- [x] Low risk: 2 devs MUST perform testing as specified above and attach their results as comments to this PR **before** merging.
- [ ] High risk: QA team MUST perform additional testing in the specified affected areas **before** merging.

Worsct case scenario is the ctrl-K search doesn't work anymore

### CODE REVIEW

File: ui/app/mainui/AppMain.qml [hunk 1]
```diff
@@ -1737,14 +1737,25 @@ Item {
             asynchronous: true
             sourceComponent: StatusSearchListPopup {
                 searchBoxPlaceholder: qsTr("Where do you want to go?")
-                model: rootStore.chatSearchModel
+                model: SortFilterProxyModel {
+                    sourceModel: rootStore.chatSearchModel
+
+                    filters: [
+                        FastExpressionFilter {
```
Comment (micieslak) [inline ui/app/mainui/AppMain.qml:1744]: Please use `AnyOf` and two `SearchFilter`s. It will be faster and cleaner than `FastExpressionFilter`.

**Review by micieslak — COMMENTED** [review]
Nice cleanup, but there is one more thing to do in order to remove the old fancy workaround :)

File: ui/imports/shared/status/StatusSearchListPopup.qml [hunk 2]
```diff
@@ -121,7 +119,6 @@ Popup {
                     onLoaded: {
                         item.modelData = delegateItem.myData
                         item.isCurrentItem = Qt.binding(() => delegateItem.ListView.isCurrentItem)
-                        delegateItem.visible = Qt.binding(() => item.filterAccepts(searchBox.text))
```
[the remark this answers is not in the document]
Comment (jrainville) [reply ui/imports/shared/status/StatusSearchListPopup.qml:124]: Oof, yeah that code was not great. All cleaned up now

**Review by micieslak — APPROVED** [review]
Looks nice! One small suggestion left in a comment.

**Pull request merged**